shell/browser/protocol_registry.h

```diff
@@ -27,7 +27,8 @@ class ProtocolRegistry {
 
   void RegisterURLLoaderFactories(
       URLLoaderFactoryType type,
-      content::ContentBrowserClient::NonNetworkURLLoaderFactoryMap* factories);
+      content::ContentBrowserClient::NonNetworkURLLoaderFactoryMap* factories,
+      content::WebContents* web_contents = nullptr);
```

Comment:
nit: I think we can [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter) this a bit and just pass a bool for whether to allow file scheme access:

```suggestion
      bool allow_file_access = false);
```

& then move the WebPreferences logic into ElectronBrowserClient::RegisterNonNetworkSubresourceURLLoaderFactories